test_runner/regress/test_tenants.py

```diff
@@ -475,3 +476,37 @@ def only_int(samples: List[Sample]) -> int:
     assert counts
     log.info(f"directory counts: {counts}")
     assert counts[2] > COUNT_AT_LEAST_EXPECTED
+
+
+def test_timelines_parallel_endpoints(neon_simple_env: NeonEnv):
+    """
+    (Relaxed) regression test for issue that led to https://github.com/neondatabase/neon/pull/9268
+    Create many endpoints in parallel and then restart them
+    """
+    env = neon_simple_env
+    pageserver_http_client = env.pageserver.http_client()
+
+    # This param needs to be 200+ to reproduce the limit issue
+    n_threads = 16
```

Comment:
If this needs to be >200 to reproduce the bug, shouldn't it default to e.g. 256 in order to detect a regression?

Reply:
there is a low chance that there is an error with bringing up the compute:
```
2024-10-12 10:46:23.67  INFO [neon_cli.py:139] Run ['/home/arpad/src/neon.main/target/debug/neon_local', 'endpoint', 'start', '--safekeepers', '1', 'ep-20'] failed:
  stdout:
    Starting existing endpoint ep-20...
    Starting postgres node at 'postgresql://cloud_admin@127.0.0.1:15054/postgres'
    SIGKILL & wait the started process
  stderr:
    command failed: unexpected compute status: Empty
```
if the number of threads is really high, the chance that it occurs is also way higher. so the test would be super flaky (instead of only a little bit flaky like right now).

Also, hyper explicitly said that they don't guarantee that the default limit is 200, so any future release might increase or decrease that limit. At that point, it's not useful IMO to put the number of computes to 256.

Reply:
Yeah, this is always going to be arbitrary. But if the regression test as configured wouldn't have caught the original problem, is it really guarding against a regression?

I suppose these kinds of issues would be better handled by running proper large-scale end-to-end tests with representative workloads, at some point.